src/interface/blas1_interface.hpp

```diff
@@ -325,6 +379,44 @@ typename ValueType<container_0_t>::type _dot(executor_t &ex, index_t _N,
   return res[0];
 }
 
+/**
+ * \brief Computes the inner product of two vectors with double precision
+ * accumulation and adds a scalar to the result (synchronous version that
+ * returns the result directly)
+ * @tparam executor_t Executor type
+ * @tparam container_0_t Buffer Iterator
+ * @tparam container_1_t Buffer Iterator
+ * @tparam container_2_t Buffer Iterator
+ * @tparam index_t Index type
+ * @tparam increment_t Increment type
+ * @param ex Executor
+ * @param _N Input buffer sizes. If size 0, the result will be sb.
+ * @param sb scalar to add to the results of the inner product.
+ * @param _vx Buffer holding input vector x
+ * @param _incx Stride of vector x
+ * @param _vy Buffer holding input vector y
+ * @param _incy Stride of vector y
+ * @param _rs output buffer
+ * @return vector of events to wait for.
+ */
+template <typename executor_t, typename container_0_t, typename container_1_t,
+          typename index_t, typename increment_t>
+typename ValueType<container_0_t>::type _sdsdot(executor_t &ex, index_t _N,
+                                                float sb, container_0_t _vx,
+                                                increment_t _incx,
+                                                container_1_t _vy,
+                                                increment_t _incy) {
+  using element_t = typename ValueType<container_0_t>::type;
+  element_t res{};
+  auto gpu_res = make_sycl_iterator_buffer<element_t>(static_cast<index_t>(1));
+  auto event1 =
+      blas::internal::_sdsdot(ex, _N, sb, _vx, _incx, _vy, _incy, gpu_res);
+  ex.get_policy_handler().wait(event1);
+  auto event2 = ex.get_policy_handler().copy_to_host(gpu_res, &res, 1);
```

Comment:
Is there a way by which the `copy_to_host` can take event1 as a dependcy argument as you can do with lots of USM functions in SYCL? If it could, the first `.wait()` could be avoided.

Reply:
There is no existing function overload that allows that. And I honestly would rather not mess with it to create a new one on this PR.

Reply:
If it is a buffer technically you don't need to wait on that as SYCL handles the dependency, if it is USM you have to pass the event list as an input so that you can explicitly wait on that.